core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java

```diff
@@ -111,12 +112,34 @@ public MssqlSqlDialect(Context context) {
       case TRIM:
         unparseTrim(writer, call, leftPrec, rightPrec);
         break;
+      case OTHER_FUNCTION:
+        unparseOtherFunction(writer, call, leftPrec, rightPrec);
+        break;
       default:
         super.unparseCall(writer, call, leftPrec, rightPrec);
       }
     }
   }
 
+  public void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "LN":
+    case "LOG10":
+      final String frameName = call.getOperator().getName()
+                                    .equalsIgnoreCase("LN") ? "LOG" : "LOG10";
+      final SqlWriter.Frame logFrame = writer.startFunCall(frameName);
+      if (call.operand(0).getKind().toString().equalsIgnoreCase("CAST")) {
```

Comment:
Is it possible to unparse 'CAST' as well ?

Reply:
Done